.github/workflows/build.yaml

```
@@ -2,7 +2,8 @@ name: Build and verify
 on:
   pull_request:
   push:
-    branches: [$default-branch]
+    branches:
+      - master
```

We could still do:
```suggestion
    branches: [ master ]
```

(Assuming the earlier intention was to match `jdk: [ 11.0.16, 17.0.4 ]` below. Or we could change both to multi-line lists. Maybe that's nicer?)

---

IMO the `jdk` field is a bit different as for `matrix` that is the way to define the values, see [here](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs). 

AFAICS that is the only way there. For branches, especially if it is only one, I prefer the `- master` way. 

WDYT?

---

The docs also use both approaches, not very consistent. `[something]` is the same as `- something`, so in the end shouldn't matter. It can be a shorthand, which are especially helpful for matrices, whereas branches are longer to begin with.

```
jdk: [ 11.0.16, 17.0.4 ]
```

instead of 

```
jdk:
 - 11.0.16
 - 17.0.4
```

Nevertheless, I don't see us expanding this list to a wide range, so using the shorthand is pretty clean.